agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/telemetry/TelemetryClient.java

```diff
@@ -409,13 +410,21 @@ public String getRoleInstance() {
   }
 
   // used during Azure Functions placeholder specialization
-  public void updateConnectionString(ConnectionString connectionString) {
-    this.connectionString = connectionString;
-  }
-
-  // used during Azure Functions placeholder specialization
-  public void updateStatsbeatConnectionString(StatsbeatConnectionString statsbeatConnectionString) {
-    this.statsbeatConnectionString = statsbeatConnectionString;
+  // and also used by Azure Spring Apps dynamic configuration
+  public void updateConnectionStrings(
+      @Nullable String connectionString,
+      @Nullable String statsbeatInstrumentationKey,
+      @Nullable String statsbeatEndpoint) {
+
+    if (Strings.isNullOrEmpty(connectionString)) {
+      this.connectionString = null;
```

Comment:
is it necessary to set it to null? 

if (!Strings.isNullOrEmpty(conenctionString)) {)) {
}

Reply:
I think this is helpful because we have a couple of places that get the connection string / ikey, and then those won't get outdated ones (Azure Spring Cloud can set the connection string to null at runtime)